KSERVE_ENABLE_SELF_SIGNED_CA does not work for "make deploy"

/kind bug

**What steps did you take and what happened:**

Calling "make deploy" with KSERVE_ENABLE_SELF_SIGNED_CA=true, and no cert-manager installed, yields:

```
# Remove the certmanager certificate if KSERVE_ENABLE_SELF_SIGNED_CA is not false
cd config/default && if [ true != false ]; then \
	/Users/roland/Development/rhods/kserve/bin/kustomize edit remove resource certmanager/certificate.yaml; \
	else /Users/roland/Development/rhods/kserve/bin/kustomize edit add resource certmanager/certificate.yaml; fi;
/Users/roland/Development/rhods/kserve/bin/kustomize build config/default | kubectl apply -f -
...
Error from server (NotFound): error when creating "STDIN": the server could not find the requested resource (post certificates.cert-manager.io)
Error from server (NotFound): error when creating "STDIN": the server could not find the requested resource (post issuers.cert-manager.io)
make: *** [deploy] Error 1
```

**What did you expect to happen:**

I would expect that `make deploy` runs through without error

Observations:

* `certmanager/certificate.yaml` does not exist in `config/default/kustomization.yml` (but `../certmanager`, so the `kustomize delete/add` command does not work
* `config/default/kustomization.yaml` references `Certificate` resources directly, those are not available if no cert-manager is installed. So updating the resource path in kustomization.yml to ../certmanager doesn't help.
* It seems to work though if cert-manager is installed and its CRDs are available.

**Environment:**

- Latest KServe `master` branch
- No cert-manager installed.

--- a/Makefile
+++ b/Makefile
@@ -75,26 +75,27 @@ run: generate fmt vet lint
 deploy: manifests
 	# Remove the certmanager certificate if KSERVE_ENABLE_SELF_SIGNED_CA is not false
 	cd config/default && if [ ${KSERVE_ENABLE_SELF_SIGNED_CA} != false ]; then \
-	${KUSTOMIZE} edit remove resource certmanager/certificate.yaml; \
-	else ${KUSTOMIZE} edit add resource certmanager/certificate.yaml; fi;
+	${KUSTOMIZE} edit remove resource ../certmanager; \
+	else ${KUSTOMIZE} edit add resource ../certmanager; fi;
 	${KUSTOMIZE} build config/default | kubectl apply -f -
+	if [ ${KSERVE_ENABLE_SELF_SIGNED_CA} != false ]; then ./hack/self-signed-ca.sh; fi;
 	kubectl wait --for=condition=ready pod -l control-plane=kserve-controller-manager -n kserve --timeout=300s
 	sleep 2
 	${KUSTOMIZE} build config/runtimes | kubectl apply -f -
-	if [ ${KSERVE_ENABLE_SELF_SIGNED_CA} != false ]; then ./hack/self-signed-ca.sh; fi;
+
 
 deploy-dev: manifests
 	./hack/image_patch_dev.sh development
 	# Remove the certmanager certificate if KSERVE_ENABLE_SELF_SIGNED_CA is not false
 	cd config/default && if [ ${KSERVE_ENABLE_SELF_SIGNED_CA} != false ]; then \
-	${KUSTOMIZE} edit remove resource certmanager/certificate.yaml; \
-	else ${KUSTOMIZE} edit add resource certmanager/certificate.yaml; fi;
+	${KUSTOMIZE} edit remove resource ../certmanager; \
+	else ${KUSTOMIZE} edit add resource ../certmanager; fi;
 	${KUSTOMIZE} build config/overlays/development | kubectl apply -f -
+	if [ ${KSERVE_ENABLE_SELF_SIGNED_CA} != false ]; then ./hack/self-signed-ca.sh; fi;
 	# TODO: Add runtimes as part of default deployment
 	kubectl wait --for=condition=ready pod -l control-plane=kserve-controller-manager -n kserve --timeout=300s
 	sleep 2
 	${KUSTOMIZE} build config/runtimes | kubectl apply -f -
-	if [ ${KSERVE_ENABLE_SELF_SIGNED_CA} != false ]; then ./hack/self-signed-ca.sh; fi;
 
 deploy-dev-sklearn: docker-push-sklearn kustomize
 	./hack/serving_runtime_image_patch.sh "kserve-sklearnserver.yaml" "${KO_DOCKER_REPO}/${SKLEARN_IMG}"

--- a/config/default/kustomization.yaml
+++ b/config/default/kustomization.yaml
@@ -8,197 +8,188 @@ namespace: kserve
 #commonLabels:
 #  app.kubernetes.io/name: kserve
 resources:
-- ../crd
-- ../configmap
-- ../rbac
-- ../manager
-- ../webhook
-- ../certmanager
+  - ../crd
+  - ../configmap
+  - ../rbac
+  - ../manager
+  - ../webhook
+  - ../certmanager
 
 generatorOptions:
   disableNameSuffixHash: true
 
 
+
 replacements:
   - source:
-      version: v1
+      fieldPath: metadata.name
       kind: Service
       name: kserve-webhook-server-service
-      fieldPath: metadata.name
-    targets:
-      - select:
-          kind: CustomResourceDefinition
-          name: inferenceservices.serving.kserve.io
-        fieldPaths:
-          - spec.conversion.webhook.clientConfig.service.name
-      - select:
-          kind: MutatingWebhookConfiguration
-          name: inferenceservice.serving.kserve.io
-        fieldPaths:
-          - webhooks.*.clientConfig.service.name
-      - select:
-          kind: ValidatingWebhookConfiguration
-          name: inferenceservice.serving.kserve.io
-        fieldPaths:
-          - webhooks.*.clientConfig.service.name
-      - select:
-          kind: ValidatingWebhookConfiguration
-          name: trainedmodel.serving.kserve.io
-        fieldPaths:
-          - webhooks.*.clientConfig.service.name
-      - select:
-          kind: ValidatingWebhookConfiguration
-          name: inferencegraph.serving.kserve.io
-        fieldPaths:
-          - webhooks.*.clientConfig.service.name
-      - select:
-          kind: ValidatingWebhookConfiguration
-          name: clusterservingruntime.serving.kserve.io
-        fieldPaths:
-          - webhooks.*.clientConfig.service.name
-      - select:
-          kind: ValidatingWebhookConfiguration
-          name: servingruntime.serving.kserve.io
-        fieldPaths:
-          - webhooks.*.clientConfig.service.name
-      - select:
-          kind: Certificate
-          name: serving-cert
-          namespace: kserve
-        fieldPaths:
-          - spec.commonName
-          - spec.dnsNames.0
-        options:
-          delimiter: '.'
-          index: 0
-
-  - source:
-      group: cert-manager.io
-      kind: Certificate
-      name: serving-cert
       version: v1
-      fieldPath: metadata.namespace
     targets:
-      - select:
+      - fieldPaths:
+          - spec.conversion.webhook.clientConfig.service.name
+        select:
           kind: CustomResourceDefinition
           name: inferenceservices.serving.kserve.io
-        fieldPaths:
-          - spec.conversion.webhook.clientConfig.service.namespace
-      - select:
+      - fieldPaths:
+          - webhooks.*.clientConfig.service.name
+        select:
           kind: MutatingWebhookConfiguration
           name: inferenceservice.serving.kserve.io
-        fieldPaths:
-          - webhooks.*.clientConfig.service.namespace
-      - select:
+      - fieldPaths:
+          - webhooks.*.clientConfig.service.name
+        select:
           kind: ValidatingWebhookConfiguration
           name: inferenceservice.serving.kserve.io
-        fieldPaths:
-          - webhooks.*.clientConfig.service.namespace
-      - select:
+      - fieldPaths:
+          - webhooks.*.clientConfig.service.name
+        select:
           kind: ValidatingWebhookConfiguration
           name: trainedmodel.serving.kserve.io
-        fieldPaths:
-          - webhooks.*.clientConfig.service.namespace
-      - select:
+      - fieldPaths:
+          - webhooks.*.clientConfig.service.name
+        select:
           kind: ValidatingWebhookConfiguration
           name: inferencegraph.serving.kserve.io
-        fieldPaths:
-          - webhooks.*.clientConfig.service.namespace
-      - select:
+      - fieldPaths:
+          - webhooks.*.clientConfig.service.name
+        select:
           kind: ValidatingWebhookConfiguration
           name: clusterservingruntime.serving.kserve.io
-        fieldPaths:
-          - webhooks.*.clientConfig.service.namespace
-      - select:
+      - fieldPaths:
+          - webhooks.*.clientConfig.service.name
+        select:
           kind: ValidatingWebhookConfiguration
           name: servingruntime.serving.kserve.io
-        fieldPaths:
-          - webhooks.*.clientConfig.service.namespace
-      - select:
-          kind: Certificate
-          name: serving-cert
-          namespace: kserve
-        fieldPaths:
+      - fieldPaths:
           - spec.commonName
           - spec.dnsNames.0
         options:
-          delimiter: '.'
-          index: 1
-      - select:
+          delimiter: .
+        select:
+          kind: Certificate
+          name: serving-cert
+          namespace: kserve
+  - source:
+      fieldPath: metadata.namespace
+      kind: Deployment
+      name: kserve-controller-manager
+      version: v1
+    targets:
+      - fieldPaths:
+          - spec.conversion.webhook.clientConfig.service.namespace
+        select:
           kind: CustomResourceDefinition
           name: inferenceservices.serving.kserve.io
-        fieldPaths:
-          - metadata.annotations.[cert-manager.io/inject-ca-from]
-        options:
-          delimiter: '/'
-          index: 0
-      - select:
+      - fieldPaths:
+          - webhooks.*.clientConfig.service.namespace
+        select:
           kind: MutatingWebhookConfiguration
           name: inferenceservice.serving.kserve.io
-        fieldPaths:
-          - metadata.annotations.[cert-manager.io/inject-ca-from]
-        options:
-          delimiter: '/'
-          index: 0
-      - select:
+      - fieldPaths:
+          - webhooks.*.clientConfig.service.namespace
+        select:
           kind: ValidatingWebhookConfiguration
           name: inferenceservice.serving.kserve.io
-        fieldPaths:
-          - metadata.annotations.[cert-manager.io/inject-ca-from]
-        options:
-          delimiter: '/'
-          index: 0
-      - select:
+      - fieldPaths:
+          - webhooks.*.clientConfig.service.namespace
+        select:
           kind: ValidatingWebhookConfiguration
           name: trainedmodel.serving.kserve.io
-        fieldPaths:
-          - metadata.annotations.[cert-manager.io/inject-ca-from]
-        options:
-          delimiter: '/'
-          index: 0
-      - select:
+      - fieldPaths:
+          - webhooks.*.clientConfig.service.namespace
+        select:
           kind: ValidatingWebhookConfiguration
           name: inferencegraph.serving.kserve.io
-        fieldPaths:
-          - metadata.annotations.[cert-manager.io/inject-ca-from]
-        options:
-          delimiter: '/'
-          index: 0
-      - select:
+      - fieldPaths:
+          - webhooks.*.clientConfig.service.namespace
+        select:
           kind: ValidatingWebhookConfiguration
           name: clusterservingruntime.serving.kserve.io
-        fieldPaths:
-          - metadata.annotations.[cert-manager.io/inject-ca-from]
-        options:
-          delimiter: '/'
-          index: 0
-      - select:
+      - fieldPaths:
+          - webhooks.*.clientConfig.service.namespace
+        select:
           kind: ValidatingWebhookConfiguration
           name: servingruntime.serving.kserve.io
-        fieldPaths:
+      - fieldPaths:
+          - spec.commonName
+          - spec.dnsNames.0
+        options:
+          delimiter: .
+          index: 1
+        select:
+          kind: Certificate
+          name: serving-cert
+          namespace: kserve
+      - fieldPaths:
           - metadata.annotations.[cert-manager.io/inject-ca-from]
         options:
-          delimiter: '/'
-          index: 0
+          delimiter: /
+        select:
+          kind: CustomResourceDefinition
+          name: inferenceservices.serving.kserve.io
+      - fieldPaths:
+          - metadata.annotations.[cert-manager.io/inject-ca-from]
+        options:
+          delimiter: /
+        select:
+          kind: MutatingWebhookConfiguration
+          name: inferenceservice.serving.kserve.io
+      - fieldPaths:
+          - metadata.annotations.[cert-manager.io/inject-ca-from]
+        options:
+          delimiter: /
+        select:
+          kind: ValidatingWebhookConfiguration
+          name: inferenceservice.serving.kserve.io
+      - fieldPaths:
+          - metadata.annotations.[cert-manager.io/inject-ca-from]
+        options:
+          delimiter: /
+        select:
+          kind: ValidatingWebhookConfiguration
+          name: trainedmodel.serving.kserve.io
+      - fieldPaths:
+          - metadata.annotations.[cert-manager.io/inject-ca-from]
+        options:
+          delimiter: /
+        select:
+          kind: ValidatingWebhookConfiguration
+          name: inferencegraph.serving.kserve.io
+      - fieldPaths:
+          - metadata.annotations.[cert-manager.io/inject-ca-from]
+        options:
+          delimiter: /
+        select:
+          kind: ValidatingWebhookConfiguration
+          name: clusterservingruntime.serving.kserve.io
+      - fieldPaths:
+          - metadata.annotations.[cert-manager.io/inject-ca-from]
+        options:
+          delimiter: /
+        select:
+          kind: ValidatingWebhookConfiguration
+          name: servingruntime.serving.kserve.io
 
-  # Protect the /metrics endpoint by putting it behind auth.
-  # Only one of manager_auth_proxy_patch.yaml and
-  # manager_prometheus_metrics_patch.yaml should be enabled.
-  # If you want your controller-manager to expose the /metrics
-  # endpoint w/o any authn/z, uncomment the following line and
-  # comment manager_auth_proxy_patch.yaml.
-  # Only one of manager_auth_proxy_patch.yaml and
-  # manager_prometheus_metrics_patch.yaml should be enabled.
-  #- manager_prometheus_metrics_patch.yaml
+    # Protect the /metrics endpoint by putting it behind auth.
+    # Only one of manager_auth_proxy_patch.yaml and
+    # manager_prometheus_metrics_patch.yaml should be enabled.
+    # If you want your controller-manager to expose the /metrics
+    # endpoint w/o any authn/z, uncomment the following line and
+    # comment manager_auth_proxy_patch.yaml.
+    # Only one of manager_auth_proxy_patch.yaml and
+    # manager_prometheus_metrics_patch.yaml should be enabled.
+    #- manager_prometheus_metrics_patch.yaml
 patches:
-- path: manager_image_patch.yaml
-- path: manager_auth_proxy_patch.yaml
-- path: isvc_mutatingwebhook_cainjection_patch.yaml
-- path: isvc_validatingwebhook_cainjection_patch.yaml
-- path: inferencegraph_validatingwebhook_cainjection_patch.yaml
-- path: trainedmodel_validatingwebhook_cainjection_patch.yaml
-- path: clusterservingruntime_validatingwebhook_cainjection_patch.yaml
-- path: servingruntime_validationwebhook_cainjection_patch.yaml
-- path: manager_resources_patch.yaml
-- path: inferenceservice_conversion_webhook.yaml
-- path: cainjection_conversion_webhook.yaml
+  - path: manager_image_patch.yaml
+  - path: manager_auth_proxy_patch.yaml
+  - path: isvc_mutatingwebhook_cainjection_patch.yaml
+  - path: isvc_validatingwebhook_cainjection_patch.yaml
+  - path: inferencegraph_validatingwebhook_cainjection_patch.yaml
+  - path: trainedmodel_validatingwebhook_cainjection_patch.yaml
+  - path: clusterservingruntime_validatingwebhook_cainjection_patch.yaml
+  - path: servingruntime_validationwebhook_cainjection_patch.yaml
+  - path: manager_resources_patch.yaml
+  - path: inferenceservice_conversion_webhook.yaml
+  - path: cainjection_conversion_webhook.yaml

--- a/config/overlays/kubeflow/kustomization.yaml
+++ b/config/overlays/kubeflow/kustomization.yaml
@@ -22,11 +22,10 @@ configurations:
 
 replacements:
   - source:
-      group: cert-manager.io
-      kind: Certificate
-      name: serving-cert
-      version: v1
       fieldPath: metadata.namespace
+      kind: Deployment
+      name: kserve-controller-manager
+      version: v1
     targets:
       - select:
           kind: CustomResourceDefinition

--- a/hack/self-signed-ca.sh
+++ b/hack/self-signed-ca.sh
@@ -1,9 +1,24 @@
 #!/bin/bash
 
-set -e
+# Copyright 2023 The KServe Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o pipefail
 
 usage() {
-    cat <<EOF
+  cat <<EOF
 Generate certificate suitable for use with an KFServing webhook service.
 This script uses openssl to generate self-signed CA certificate that is
 suitable for use with KFServing webhook services. See
@@ -13,63 +28,72 @@ The server key/cert CA cert are stored in a k8s secret.
 
 usage: ${0} [OPTIONS]
 The following flags are optional.
-       --service           Service name of webhook. Default: kserve-webhook-server-service
-       --namespace         Namespace where webhook service and secret reside. Default: kserve
-       --secret            Secret name for CA certificate and server certificate/key pair. Default: kserve-webhook-server-cert
-       --webhookName       Name for the mutating and validating webhook config. Default: inferenceservice.serving.kserve.io
-       --webhookDeployment Statefulset name of the webhook controller. Default: kserve-controller-manager
+       --service                Service name of webhook. Default: kserve-webhook-server-service
+       --namespace              Namespace where webhook service and secret reside. Default: kserve
+       --secret                 Secret name for CA certificate and server certificate/key pair. Default: kserve-webhook-server-cert
+       --mutatingWebhookName    Name for the mutating webhook config. This can be specified multiple times. Default: inferenceservice.serving.kserve.io
+       --validatingWebhookName  Name for the validating webhook config. This can be specified multiple times. Default: All validating webhooks
+       --webhookDeployment      Deployment name of the webhook controller. Default: kserve-controller-manager
 EOF
-    exit 1
+  exit 1
 }
-
+mutatingWebhookNames=()
+validatingWebhookNames=()
 while [[ $# -gt 0 ]]; do
-    case ${1} in
-        --service)
-            service="$2"
-            shift
-            ;;
-        --secret)
-            secret="$2"
-            shift
-            ;;
-        --namespace)
-            namespace="$2"
-            shift
-            ;;
-        --webhookName)
-            webhookName="$2"
-            shift
-            ;;
-        --webhookDeployment)
-            webhookDeployment="$2"
-            shift
-            ;;
-        *)
-            usage
-            ;;
-    esac
+  case ${1} in
+  --service)
+    service="$2"
     shift
+    ;;
+  --secret)
+    secret="$2"
+    shift
+    ;;
+  --namespace)
+    namespace="$2"
+    shift
+    ;;
+  --mutatingWebhookName)
+    mutatingWebhookNames+=("$2")
+    shift
+    ;;
+  --validatingWebhookName)
+    validatingWebhookNames+=("$2")
+    shift
+    ;;
+  --webhookDeployment)
+    webhookDeployment="$2"
+    shift
+    ;;
+  *)
+    usage
+    ;;
+  esac
+  shift
 done
 #TODO check backward compatibility
-[ -z ${secret} ] && secret=kserve-webhook-server-cert
-[ -z ${namespace} ] && namespace=kserve
-[ -z ${webhookDeployment} ] && webhookDeployment=kserve-controller-manager
-[ -z ${webhookName} ] && webhookName=inferenceservice.serving.kserve.io
-[ -z ${service} ] && service=kserve-webhook-server-service
-webhookDeploymentName=${webhookDeployment}-0
-webhookConfigName=${webhookName}
-echo service: ${service}
-echo namespace: ${namespace}
-echo secret: ${secret}
-echo webhookDeploymentName: ${webhookDeploymentName}
-echo webhookConfigName: ${webhookConfigName}
+[ -z "${secret}" ] && secret=kserve-webhook-server-cert
+[ -z "${namespace}" ] && namespace=kserve
+[ -z "${webhookDeployment}" ] && webhookDeployment=kserve-controller-manager
+[ "${#validatingWebhookNames[@]}" -eq 0 ] && validatingWebhookNames=("inferenceservice.serving.kserve.io" "inferencegraph.serving.kserve.io" "servingruntime.serving.kserve.io" "clusterservingruntime.serving.kserve.io" "trainedmodel.serving.kserve.io")
+[ "${#mutatingWebhookNames[@]}" -eq 0 ] && mutatingWebhookNames=("inferenceservice.serving.kserve.io")
+[ -z "${service}" ] && service=kserve-webhook-server-service
+webhookDeploymentName=${webhookDeployment}
+mutatingWebhookConfigNames=("${mutatingWebhookNames[@]}")
+validatingWebhookConfigNames=("${validatingWebhookNames[@]}")
+echo service: "${service}"
+echo namespace: "${namespace}"
+echo secret: "${secret}"
+echo webhookDeploymentName: "${webhookDeploymentName}"
+echo mutatingWebhookConfigNames: "${mutatingWebhookConfigNames[@]}"
+echo validatingWebhookConfigNames: "${validatingWebhookConfigNames[@]}"
 if [ ! -x "$(command -v openssl)" ]; then
-    echo "openssl not found"
-    exit 1
+  echo "openssl not found"
+  exit 1
 fi
 tmpdir=$(mktemp -d)
 echo "creating certs in tmpdir ${tmpdir} "
-cat <<EOF >> ${tmpdir}/csr.conf
+cat <<EOF >>"${tmpdir}"/csr.conf
 [req]
 req_extensions = v3_req
 distinguished_name = req_distinguished_name
@@ -88,72 +112,76 @@ DNS.5 = ${service}.${namespace}.svc.cluster.local
 
 EOF
 # Create CA and Server key/certificate
-openssl genrsa -out ${tmpdir}/ca.key 2048
-openssl req -x509 -newkey rsa:2048 -key ${tmpdir}/ca.key -out ${tmpdir}/ca.crt -days 365 -nodes -subj "/CN=${service}.${namespace}.svc"
+openssl genrsa -out "${tmpdir}"/ca.key 2048
+openssl req -x509 -newkey rsa:2048 -key "${tmpdir}"/ca.key -out "${tmpdir}"/ca.crt -days 365 -nodes -subj "/CN=${service}.${namespace}.svc"
 
-openssl genrsa -out ${tmpdir}/server.key 2048
-openssl req -new -key ${tmpdir}/server.key -subj "/CN=${service}.${namespace}.svc" -out ${tmpdir}/server.csr -config ${tmpdir}/csr.conf
+openssl genrsa -out "${tmpdir}"/server.key 2048
+openssl req -new -key "${tmpdir}"/server.key -subj "/CN=${service}.${namespace}.svc" -out ${tmpdir}/server.csr -config ${tmpdir}/csr.conf
 
 # Self sign
-openssl x509 -extensions v3_req -req -days 365 -in ${tmpdir}/server.csr -CA ${tmpdir}/ca.crt -CAkey ${tmpdir}/ca.key -CAcreateserial -out ${tmpdir}/server.crt -extfile ${tmpdir}/csr.conf
+openssl x509 -extensions v3_req -req -days 365 -in "${tmpdir}"/server.csr -CA "${tmpdir}"/ca.crt -CAkey "${tmpdir}"/ca.key -CAcreateserial -out "${tmpdir}"/server.crt -extfile "${tmpdir}"/csr.conf
 # create the secret with server cert/key
 kubectl create secret generic ${secret} \
-        --from-file=tls.key=${tmpdir}/server.key \
-        --from-file=tls.crt=${tmpdir}/server.crt \
-        --dry-run -o yaml |
-    kubectl -n ${namespace} apply -f -
+  --from-file=tls.key="${tmpdir}"/server.key \
+  --from-file=tls.crt="${tmpdir}"/server.crt \
+  --dry-run=client -o yaml |
+  kubectl -n ${namespace} apply -f -
 # Webhook pod needs to be restarted so that the service reload the secret
-# http://github.com/kueflow/kubeflow/issues/3227
-webhookPod=$(kubectl get pods -n ${namespace} |grep ${webhookDeploymentName} |awk '{print $1;}')
+# http://github.com/kubeflow/kubeflow/issues/3227
+webhookPod=$(kubectl get pods -n ${namespace} | grep ${webhookDeploymentName} | awk '{print $1;}')
 # ignore error if webhook pod does not exist
-kubectl delete pod ${webhookPod} -n ${namespace} 2>/dev/null || true
+kubectl delete pod "${webhookPod}" -n "${namespace}" 2>/dev/null || true
 echo "webhook ${webhookPod} is restarted to utilize the new secret"
 
 echo "CA Certificate:"
-cat ${tmpdir}/ca.crt
+cat "${tmpdir}"/ca.crt
 
 # -a means base64 encode
-caBundle=$(cat ${tmpdir}/ca.crt | openssl enc -a -A)
+caBundle=$(cat "${tmpdir}"/ca.crt | openssl enc -a -A)
 echo "Encoded CA:"
 echo -e "${caBundle} \n"
 
 # check if jq is installed
 if [ ! -x "$(command -v jq)" ]; then
-    echo "jq not found"
-    exit 1
+  echo "jq not found"
+  exit 1
 fi
-# Patch CA Certificate to mutatingWebhook
-mutatingWebhookCount=$(kubectl get mutatingwebhookconfiguration ${webhookConfigName} -ojson | jq -r '.webhooks' | jq length)
-# build patchstring based on webhook counts
-mutatingPatchString='['
-for i in $(seq 0 $(($mutatingWebhookCount-1)))
-do
+for webhookConfigName in "${mutatingWebhookConfigNames[@]}"; do
+  echo "Patching mutating webhook ${webhookConfigName}"
+  # Patch CA Certificate to mutatingWebhook
+  mutatingWebhookCount=$(kubectl get mutatingwebhookconfiguration "${webhookConfigName}" -ojson | jq -r '.webhooks' | jq length)
+  # build patchstring based on webhook counts
+  mutatingPatchString='['
+  for i in $(seq 0 $(($mutatingWebhookCount - 1))); do
     mutatingPatchString=$mutatingPatchString'{"op": "replace", "path": "/webhooks/'$i'/clientConfig/caBundle", "value":"{{CA_BUNDLE}}"}, '
-done
-# strip ', '
-mutatingPatchString=${mutatingPatchString%, }']'
-mutatingPatchString=$(echo ${mutatingPatchString} | sed "s|{{CA_BUNDLE}}|${caBundle}|g")
+  done
+  # strip ', '
+  mutatingPatchString=${mutatingPatchString%, }']'
+  mutatingPatchString=$(echo ${mutatingPatchString} | sed "s|{{CA_BUNDLE}}|${caBundle}|g")
 
-echo "patching ca bundle for mutating webhook configuration..."
-kubectl patch mutatingwebhookconfiguration ${webhookConfigName} \
+  echo "patching ca bundle for mutating webhook configuration..."
+  kubectl patch mutatingwebhookconfiguration "${webhookConfigName}" \
     --type='json' -p="${mutatingPatchString}"
-
-# Patch CA Certificate to validatingWebhook
-validatingWebhookCount=$(kubectl get validatingwebhookconfiguration ${webhookConfigName} -ojson | jq -r '.webhooks' | jq length)
-validatingPatchString='['
-for i in $(seq 0 $(($validatingWebhookCount-1)))
-do
-    validatingPatchString=$validatingPatchString'{"op": "replace", "path": "/webhooks/'$i'/clientConfig/caBundle", "value":"{{CA_BUNDLE}}"}, '
 done
-validatingPatchString=${validatingPatchString%, }']'
-validatingPatchString=$(echo ${validatingPatchString} | sed "s|{{CA_BUNDLE}}|${caBundle}|g")
 
-echo "patching ca bundle for validating webhook configuration..."
-kubectl patch validatingwebhookconfiguration ${webhookConfigName} \
+for webhookConfigName in "${validatingWebhookConfigNames[@]}"; do
+  echo "Patching validating webhook ${webhookConfigName}"
+  # Patch CA Certificate to validatingWebhook
+  validatingWebhookCount=$(kubectl get validatingwebhookconfiguration "${webhookConfigName}" -ojson | jq -r '.webhooks' | jq length)
+  validatingPatchString='['
+  for i in $(seq 0 $(($validatingWebhookCount - 1))); do
+    validatingPatchString=$validatingPatchString'{"op": "replace", "path": "/webhooks/'$i'/clientConfig/caBundle", "value":"{{CA_BUNDLE}}"}, '
+  done
+  validatingPatchString=${validatingPatchString%, }']'
+  validatingPatchString=$(echo ${validatingPatchString} | sed "s|{{CA_BUNDLE}}|${caBundle}|g")
+
+  echo "patching ca bundle for validating webhook configuration..."
+  kubectl patch validatingwebhookconfiguration "${webhookConfigName}" \
     --type='json' -p="${validatingPatchString}"
+done
 
 echo "patching ca bundler for conversion webhook configuration.."
 conversionPatchString='[{"op": "replace", "path": "/spec/conversion/webhook/clientConfig/caBundle", "value":"{{CA_BUNDLE}}"}]'
 conversionPatchString=$(echo ${conversionPatchString} | sed "s|{{CA_BUNDLE}}|${caBundle}|g")
 kubectl patch CustomResourceDefinition inferenceservices.serving.kserve.io \
-    --type='json' -p="${conversionPatchString}"
+  --type='json' -p="${conversionPatchString}"

